upload_components/upload.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+IFS=';' read -ra DIRECTORIES <<<"${COMPONENTS_DIRECTORIES:-.}"
+NAMESPACE=${COMPONENTS_NAMESPACE:-espressif}
```

Comment:
Can the defaults be specified in one place only (either action.yml or upload.sh)?